File: 자릿수더하기/java/20221018/src/Solution.java
import java.util.stream.IntStream;

public class Solution {
    public int solution(int n) {
        int answer = addDigits(n);

        return answer;
    }

    public int addDigits(int n) {
        String number = String.valueOf(n);
        return IntStream.range(0, number.length())
                .reduce(0, (accumulator, i) -> accumulator + Integer.parseInt(number.substring(i, i + 1)));
    }
}

File: 자릿수더하기/javascript/20221018/src/solution.js
// eslint-disable-next-line import/prefer-default-export
export default function addDigits(n) {
  return [...`${n}`].map(Number).reduce((a, b) => a + b);
}

function solution(n) {
  const answer = addDigits(n);

  return answer;
}
